code/modules/reagents/chemistry/holder.dm

```diff
@@ -519,13 +519,18 @@
 	var/list/cached_reagents = reagent_list
 	if (!target)
 		return
-	if (!target.reagents || src.total_volume<=0 || !src.get_reagent_amount(reagent))
+
+	var/datum/reagents/R
+	if(istype(target, /datum/reagents))
+		R = target
+	else if(target.reagents && src.total_volume>0 && src.get_reagent_amount(reagent))
+		R = target.reagents
```

Comment:
* Please use more informative names than `R`.
* `src` is implied. Avoid it unless there's a local variable with the same name.
* Try to use spaces between variables, operators and values, for clarity.